YoutubeExplode/YoutubeClient.MediaStream.cs

```diff
@@ -12,49 +12,10 @@ namespace YoutubeExplode
     public partial class YoutubeClient
     {
         /// <inheritdoc />
-        public async Task<MediaStream> GetMediaStreamAsync(MediaStreamInfo info)
+        public Task<MediaStream> GetMediaStreamAsync(MediaStreamInfo info)
         {
             info.GuardNotNull(nameof(info));
-
-            // Maximum segment stream size
-            const long segmentSize = 9_898_989; // this number was carefully devised through research
-
-            // Determine if stream is rate-limited
-            var isRateLimited = !Regex.IsMatch(info.Url, @"ratebypass[=/]yes");
-
-            // Determine if the stream is longer than one segment
-            var isMultiSegment = info.Size > segmentSize;
-
-            // If rate-limited and long enough - split into segments and wrap into one stream
-            if (isRateLimited && isMultiSegment)
-            {
-                // Determine segment count
-                var segmentCount = (int) Math.Ceiling(1.0 * info.Size / segmentSize);
-
-                // Create resolvers for multi stream
-                var resolvers = new List<Func<Task<Stream>>>();
-                for (var i = 0; i < segmentCount; i++)
-                {
-                    // Determine segment range
-                    var from = i * segmentSize;
-                    var to = (i + 1) * segmentSize - 1;
-
-                    // Create resolver for this segment
-                    var resolver = new Func<Task<Stream>>(() => _httpClient.GetStreamAsync(info.Url, from, to));
-                    resolvers.Add(resolver);
-                }
-
-                // Create multi stream from segment resolvers
-                var stream = new AsyncMultiStream(resolvers);
-                return new MediaStream(info, stream);
-            }
-            // If not rate-limited or not long enough - get the stream in one segment
-            else
-            {
-                // Get the whole stream as one segment
-                var stream = await _httpClient.GetStreamAsync(info.Url, 0, info.Size).ConfigureAwait(false);
-                return new MediaStream(info, stream);
-            }
+            return Task.FromResult(new MediaStream(info, new YoutubeVideoStream(_httpClient, info.Url, info.Size)));
```

Comment:
I still think it would be nice if this was only used for rate-limited streams because otherwise it's a huge unnecessary overhead

Reply:
If I leave non rate-limited stream as it was, we won't be able to seek it, and I thought that additional overhead is not that big (additional class and couple if statements, which some of them I removed here), it will still call `_httpClient` once, as it did before (unless you seek it), the only difference (technically) will be that instead of calling `await _httpClient.GetStreamAsync(info.Url, 0, info.Size)` it will call `await _httpClient.GetStreamAsync(info.Url, 0, MaxSegmentSize)`. What do you think?

Reply:
I'm not sure I understand what do you mean by "it will still call `_httpClient` once". If the (non-throttled) stream has the length of 10 * MaxSegmentSize, won't that require 10 GET requests? I guess a simple solution would be to make segment size a constructor parameter and set it to `int.MaxValue` for non-throttled streams.

Reply:
Oh, I wasn't aware that non-throttled connection can have size bigger then MaxSegmentSize. I changed my code accordingly.